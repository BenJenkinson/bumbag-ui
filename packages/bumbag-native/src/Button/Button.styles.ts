import { Platform, TouchableOpacity } from 'react-native';

import { styled } from '../styled';
import { borderRadius, fontSize, fontWeight, palette, space, theme } from '../utils/theme';
import { Icon } from '../Icon/Icon';
import { Spinner } from '../Spinner/Spinner';
import { Text } from '../Text';

export const isInteractive = (props) =>
  !props.isStatic && !props.isLoading && !props.disabled && props.variant !== 'link';

export const StyledButton = styled(TouchableOpacity)`
  align-items: center;
  background-color: ${(props: any) => palette(props.palette)(props)};
  border-radius: ${borderRadius('default')};
  justify-content: center;
  flex-direction: row;
  min-height: ${(props) => `${space(5.5, 'major')(props)}px`};
  padding: 0 ${(props) => `${space(4)(props)}px`};
  text-decoration: none;

  ${(props: any) =>
    props.palette === 'default'
      ? `
          border: 1px solid ${palette('white900', { dark: 'gray600' })(props)};
        `
      : ''}

  ${theme('native.Button', 'styles.base')};

  ${(props: any) => (props.disabled ? getDisabledButtonProperties(props) : '')};
  ${(props: any) => (props.disabled ? theme('native.Button', `styles.disabled`)(props) : '')};

  ${(props: any) => (props.size ? getButtonSizeProperties(props) : '')};
  ${(props: any) => (props.size ? getButtonSizeOverrides(props) : '')};

  ${(props: any) => (props.isLoading ? getLoadingButtonProperties(props) : '')};
  ${(props) => (props.isLoading ? theme('native.Button', `styles.loading`)(props) : '')}

  ${(props: any) => (props.isStatic ? getStaticButtonProperties(props) : '')};
  ${(props) => (props.isStatic ? theme('native.Button', `styles.static`)(props) : '')}

  ${(props: any) => (isInteractive(props) ? getInteractiveButtonProperties(props) : '')};
  ${(props: any) => (isInteractive(props) ? getInteractiveButtonOverrides(props) : '')};

  ${(props: any) => (props.variant === 'outlined' ? getOutlinedButtonProperties(props) : '')};
  ${(props) => (props.variant === 'outlined' ? theme('native.Button', `styles.outlined`)(props) : '')}

  ${(props: any) => (props.variant === 'link' ? getLinkButtonProperties(props) : '')};
  ${(props) => (props.variant === 'link' ? theme('native.Button', `styles.link`)(props) : '')}

  ${(props: any) => (props.variant === 'ghost' ? getGhostButtonProperties(props) : '')};
  ${(props) => (props.variant === 'ghost' ? theme('native.Button', `styles.ghost`)(props) : '')}
`;

export const getDisabledButtonProperties = (props: any) => `
    opacity: ${props.colorMode === 'dark' ? '0.5' : '0.7'};

  ${
    Platform.OS === 'web'
      ? `
          cursor: not-allowed;
          pointer-events: unset;
        `
      : ''
  }
`;

export const getButtonSizeProperties = (props) => {
  const styles = {
    small: `
      min-height: ${space(8)(props)}px;
      padding: 0 ${space(3)(props)}px;
    `,
    default: ``,
    medium: `
      min-height: ${space(12)(props)}px;
      padding: 0 ${space(5)(props)}px;
    `,
    large: `
      min-height: ${space(13)(props)}px;
      padding: 0 ${space(6)(props)}px;
    `,
  };
  return styles[props.size];
};

export const getButtonSizeOverrides = (props) => {
  const styles = {
    small: theme('native.Button', `styles.sizes.small`)(props),
    default: theme('native.Button', `styles.sizes.default`)(props),
    medium: theme('native.Button', `styles.sizes.medium`)(props),
    large: theme('native.Button', `styles.sizes.large`)(props),
  };
  return styles[props.size];
};

export const getLoadingButtonProperties = (props: any) => `
  ${
    Platform.OS === 'web'
      ? `
          positon: relative;
          cursor: not-allowed;
        `
      : ''
  }
`;

export const getStaticButtonProperties = (props: any) => `
  ${
    Platform.OS === 'web'
      ? `
          cursor: default;
        `
      : ''
  }
`;

export const getInteractiveButtonProperties = (props: any) => `
  ${
    props.hover && props.variant !== 'link'
      ? `
          background-color: ${palette(`${props.palette === 'default' ? 'white' : props.palette}600`, {
            dark: `${props.palette === 'default' ? 'black100' : `${props.palette}600`}`,
          })(props)};
        `
      : ''
  }

  ${
    props.hoveractive && props.variant !== 'link'
      ? `
          background-color: ${palette(props.palette === 'default' ? 'white800' : `${props.palette}700`, {
            dark: `${props.palette === 'default' ? 'black200' : `${props.palette}700`}`,
          })(props)};
        `
      : ''
  }
`;

export const getInteractiveButtonOverrides = (props: any) => {
  if (props.hover && props.variant !== 'link') {
    return theme('native.Button', 'styles.hover')(props);
  }
  if (props.hoveractive && props.variant !== 'link') {
    return theme('native.Button', 'styles.hoveractive')(props);
  }
  return '';
};

export const getOutlinedButtonProperties = (props) => `
  background-color: ${palette('default')(props)};
  border: 1px solid ${palette(props.palette, { dark: `${props.palette}300` })(props)};

  ${
    props.hover && isInteractive(props)
      ? `
          background-color: ${palette(`${props.palette}Tint`, { dark: `${props.palette}Shade` })(props)};
          border: 1px solid ${palette(props.palette, { dark: `${props.palette}300` })(props)};
        `
      : ''
  }

  ${
    props.hoveractive && isInteractive(props)
      ? `
          background-color: ${palette(`${props.palette}100`, { dark: `${props.palette}Shade` })(props)};
          border: 1px solid ${palette(props.palette, { dark: `${props.palette}300` })(props)};
        `
      : ''
  }
`;

export const getGhostButtonProperties = (props) => `
  background-color: transparent;
  border: transparent;

  ${
    props.hover
      ? `
          background-color: rgba(0, 0, 0, 0.05);
        `
      : ''
  }

  ${
    props.hoveractive
      ? `
          background-color: rgba(0, 0, 0, 0.1);
        `
      : ''
  }
`;

export const getLinkButtonProperties = (props) => `
  background-color: transparent;
  border: transparent;

  ${
    props.hover && Platform.OS === 'web'
      ? `
          text-decoration: underline;
          text-decoration-color: ${
            props.palette === 'default'
              ? palette('text')(props)
              : palette(props.palette, { dark: `${props.palette}300` })(props)
          };
        `
      : ''
  }
`;

////////////////////////////////////////////////////////////////////////////////////

export const StyledButtonText = styled(Text)`
  color: ${(props: any) => palette(`${props.palette}Inverted`)(props)};
  font-weight: ${fontWeight('semibold')};

  ${theme('native.Button', 'Text.styles.base')};

  ${(props: any) => props.size && getButtonTextSizeProperties(props)};
  ${(props: any) => isInteractive(props) && getInteractiveButtonTextProperties(props)};

  ${(props: any) => props.variant === 'outlined' && getOutlinedButtonTextProperties(props)};
  ${(props: any) => props.variant === 'link' && getLinkButtonTextProperties(props)};
  ${(props: any) => props.variant === 'ghost' && getGhostButtonTextProperties(props)};
`;

export const getButtonTextSizeProperties = (props) => {
  const styles = {
    small: `
      font-size: ${fontSize('100')(props)}px;

      ${theme('native.Button', `Text.styles.sizes.small`)(props)};
    `,
    default: `
      ${theme('native.Button', `Text.styles.sizes.default`)(props)};
    `,
    medium: `
      ${theme('native.Button', `Text.styles.sizes.medium`)(props)};
    `,
    large: `
      font-size: ${fontSize('300')(props)}px;

      ${theme('native.Button', `Text.styles.sizes.large`)(props)};
    `,
  };
  return styles[props.size];
};

export const getInteractiveButtonTextProperties = (props: any) => `
  ${
    props.hover && props.variant !== 'link'
      ? `
          ${theme('native.Button', 'Text.styles.hover')(props)};
        `
      : ''
  }

  ${
    props.hoveractive && props.variant !== 'link'
      ? `
          ${theme('native.Button', 'Text.styles.hoveractive')(props)};
        `
      : ''
  }
`;

export const getOutlinedButtonTextProperties = (props) => `
  color: ${palette(props.palette, { dark: `${props.palette}300` })(props)};

  ${theme('native.Button', `Text.styles.outlined`)(props)};
`;

export const getGhostButtonTextProperties = (props) => `
  color: ${
    props.palette === 'default'
      ? palette('defaultInverted')(props)
      : palette(props.palette, { dark: `${props.palette}300` })(props)
  };

  ${theme('native.Button', `Text.styles.ghost`)(props)};
`;

export const getLinkButtonTextProperties = (props) => `
  color: ${
    props.palette === 'default'
      ? palette('text')(props)
      : palette(props.palette, { dark: `${props.palette}300` })(props)
  };

  ${
    Platform.OS !== 'web'
      ? `
          text-decoration: underline;
          text-decoration-color: ${
            props.palette === 'default'
              ? palette('text')(props)
              : palette(props.palette, { dark: `${props.palette}300` })(props)
          };
        `
      : ''
  }


  ${theme('native.Button', `Text.styles.link`)(props)};
`;

////////////////////////////////////////////////////////

export const StyledButtonIcon = styled(Icon)`
  ${(props) =>
    props.isBefore
      ? `
      margin-left: -${space(1)(props)}px;
      margin-right: ${space(2)(props)}px;
    `
      : ''};
  ${(props) => (props.isBefore ? theme('native.Button.Icon', `styles.before`)(props) : '')}

  ${(props) =>
    props.isAfter
      ? `
      margin-left: ${space(2)(props)}px;
      margin-right: -${space(1)(props)}px;
    `
      : ''};
  ${(props) => (props.isAfter ? theme('native.Button.Icon', `styles.after`)(props) : '')}

  ${(props) => theme('native.Button.Icon', `styles.base`)(props)};
`;

export const getButtonIconColor = (props) => {
  if (props.variant === 'ghost') {
    return props.palette === 'default'
      ? palette('defaultInverted')(props)
      : palette(props.palette, { dark: `${props.palette}300` })(props);
  }
  if (props.variant === 'outlined') {
    return palette(props.palette, { dark: `${props.palette}300` })(props);
  }
  if (props.variant === 'link') {
    return props.palette === 'default'
      ? palette('text')(props)
      : palette(props.palette, { dark: `${props.palette}300` })(props);
  }
  return palette(`${props.palette}Inverted`)(props);
};

////////////////////////////////////////////////////////////////////////////////////

export const ButtonSpinner = styled(Spinner)`
  position: absolute;

  ${(props) => theme('native.Button.Spinner', `styles.base`)(props)};
`;
