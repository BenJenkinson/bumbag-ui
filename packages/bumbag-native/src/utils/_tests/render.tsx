import * as React from 'react';
import { ThemeConfig } from '../../types/theme';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (
  Component: any,
  { colorMode = 'default', theme = {} }: { colorMode?: string; theme?: ThemeConfig } = {}
) =>
  render(
    <Provider platform="native" theme={theme} colorMode={colorMode}>
      {Component}
    </Provider>
  );
