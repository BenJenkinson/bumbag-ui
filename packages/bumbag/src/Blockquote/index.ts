import * as styles from './styles';

export * from './Blockquote';
export { styles as blockquoteStyles };
