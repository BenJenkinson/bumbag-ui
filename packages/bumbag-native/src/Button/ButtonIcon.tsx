import * as React from 'react';
import { createComponent, createElement, createHook, useTheme } from 'bumbag/utils';

import { Box, BoxProps } from '../Box';
import { ButtonContext, ButtonProps } from '../Button';
import { IconProps } from '../Icon';
import * as styles from './Button.styles';

export type LocalButtonIconProps = {
  color?: ButtonProps['color'];
  palette?: ButtonProps['palette'];
  size?: ButtonProps['size'];
  variant?: ButtonProps['variant'];
};
export type ButtonIconProps = BoxProps & IconProps & LocalButtonIconProps;

const useProps = createHook<ButtonIconProps>(
  (props) => {
    const { theme } = useTheme();
    const buttonProps = React.useContext(ButtonContext);
    const boxProps = Box.useProps(props);
    return {
      ...buttonProps,
      color: props.color || styles.getButtonIconColor({ ...buttonProps, theme }),
      size: '200',
      ...boxProps,
    };
  },
  { themeKey: 'native.ButtonIcon' }
);

export const ButtonIcon = createComponent<ButtonIconProps>(
  (props) => {
    const htmlProps = useProps(props);
    return createElement({
      children: props.children,
      component: styles.StyledButtonIcon,
      htmlProps,
    });
  },
  {
    attach: {
      useProps,
      displayName: 'native.ButtonIcon',
    },
    themeKey: 'native.ButtonIcon',
  }
);
