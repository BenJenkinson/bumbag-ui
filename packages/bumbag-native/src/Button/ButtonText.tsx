import * as React from 'react';
import { createComponent, createElement, createHook } from 'bumbag/utils';

import { Box, BoxProps } from '../Box';
import { ButtonContext, ButtonProps } from '../Button';
import { TextProps } from '../Text';
import * as styles from './Button.styles';

export type LocalButtonTextProps = {
  color?: ButtonProps['color'];
  palette?: ButtonProps['palette'];
  size?: ButtonProps['size'];
  variant?: ButtonProps['variant'];
};
export type ButtonTextProps = BoxProps & TextProps & LocalButtonTextProps;

const useProps = createHook<ButtonTextProps>(
  (props) => {
    const buttonProps = React.useContext(ButtonContext);
    const boxProps = Box.useProps(props);
    return { ...buttonProps, ...boxProps };
  },
  { themeKey: 'native.ButtonText' }
);

export const ButtonText = createComponent<ButtonTextProps>(
  (props) => {
    const htmlProps = useProps(props);
    return createElement({
      children: props.children,
      component: styles.StyledButtonText,
      htmlProps,
    });
  },
  {
    attach: {
      useProps,
      displayName: 'native.ButtonText',
    },
    themeKey: 'native.ButtonText',
  }
);
