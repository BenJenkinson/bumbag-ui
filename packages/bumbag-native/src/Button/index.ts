import * as buttonStyles from './Button.styles';

export * from './Button';
export { buttonStyles };
