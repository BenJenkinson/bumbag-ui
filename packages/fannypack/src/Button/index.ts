import * as styles from './styles';
import { Button as _Button } from './Button';
import { ButtonClose } from './ButtonClose';

export * from './Button';
export const Button = Object.assign(_Button, {
  Close: ButtonClose,
});
export { styles as buttonStyles };
