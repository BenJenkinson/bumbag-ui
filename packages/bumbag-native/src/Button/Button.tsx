import * as React from 'react';
import { Platform, TouchableOpacityProps as RNTouchableOpacityProps } from 'react-native';
import ConditionalWrap from 'conditional-wrap';
import { createComponent, createElement, createHook } from 'bumbag/utils';
import { ButtonType, Omit, Size, Palette } from 'bumbag/types';
import _throttle from 'lodash/throttle';

import { Box, BoxProps } from '../Box';
import { IconProps } from '../Icon';
import { SpinnerProps } from '../Spinner/Spinner';

import * as styles from './Button.styles';
import { ButtonIcon } from './ButtonIcon';
import { ButtonText } from './ButtonText';

export type LocalButtonProps = {
  /** Icon that appears on the right side of the button. */
  iconAfter?: IconProps['icon'];
  iconAfterProps?: Omit<IconProps, 'icon'>;
  /** Icon that appears on the left side of the button. */
  iconBefore?: IconProps['icon'];
  iconBeforeProps?: Omit<IconProps, 'icon'>;
  /** Adds a loading indicator to the button. */
  isLoading?: boolean;
  /** Makes the button not interactable. */
  isStatic?: boolean;
  palette?: Palette;
  size?: Size;
  /** Custom props for the isLoading spinner. */
  spinnerProps?: SpinnerProps;
  /** Sets a throttle for the press event (`onPress`) to be only invoked at most once every `x` milliseconds.  */
  throttle?: boolean | number;
  type?: ButtonType;
};
export type ButtonProps = BoxProps & RNTouchableOpacityProps & LocalButtonProps;

export const ButtonContext = React.createContext<LocalButtonProps>({});

const useProps = createHook<ButtonProps>(
  (props) => {
    const {
      children,
      color,
      iconAfter,
      iconAfterProps,
      iconBefore,
      iconBeforeProps,
      isLoading,
      isStatic,
      onPress,
      palette,
      size,
      spinnerProps,
      throttle,
      variant,
    } = props;

    const boxProps = Box.useProps(props);

    const activeOpacity = Platform.OS === 'web' || isStatic ? '1' : props.activeOpacity;

    let spinnerColor = palette;
    if (props.variant === 'default') {
      spinnerColor = `${props.palette}Inverted`;
    }
    if (props.palette === 'default') {
      spinnerColor = 'defaultInverted';
    }

    const contextValue = React.useMemo(
      () => ({
        color,
        palette,
        size,
        variant,
      }),
      [color, palette, size, variant]
    );

    const isString =
      typeof children === 'string' ||
      (Array.isArray(children) && !children.some?.((child) => typeof child !== 'string'));

    return {
      ...boxProps,
      onPress: throttle
        ? _throttle(onPress, throttle === true ? 2000 : throttle, { leading: true, trailing: false })
        : onPress,
      children: (
        <ButtonContext.Provider value={contextValue}>
          <React.Fragment>
            {props.isLoading && <styles.ButtonSpinner color={spinnerColor} {...spinnerProps} />}
            <ConditionalWrap condition={isLoading} wrap={(children) => <Box opacity="0">{children}</Box>}>
              <React.Fragment>
                {iconBefore && (
                  <ButtonIcon
                    // @ts-ignore
                    isBefore
                    icon={iconBefore}
                    {...iconBeforeProps}
                  />
                )}
                {isString ? (
                  // @ts-ignore
                  <ButtonText>{children}</ButtonText>
                ) : (
                  children
                )}
                {iconAfter && (
                  <ButtonIcon
                    // @ts-ignore
                    isAfter
                    icon={iconAfter}
                    {...iconAfterProps}
                  />
                )}
              </React.Fragment>
            </ConditionalWrap>
          </React.Fragment>
        </ButtonContext.Provider>
      ),
      activeOpacity,
    };
  },
  {
    defaultProps: {
      _hover: true,
      _focus: true,
      _active: true,
      _hoveractive: true,
      disabled: false,
      iconAfter: undefined,
      iconBefore: undefined,
      isLoading: false,
      isStatic: false,
      variant: 'default',
      palette: 'default',
      size: 'default',
      type: 'button',
    },
    themeKey: 'native.Button',
  }
);

export const Button = createComponent<ButtonProps>(
  (props) => {
    const htmlProps = useProps(props);
    return createElement({
      children: props.children,
      component: styles.StyledButton,
      htmlProps,
    });
  },
  {
    attach: {
      useProps,
      displayName: 'native.Button',
    },
    themeKey: 'native.Button',
  }
);
