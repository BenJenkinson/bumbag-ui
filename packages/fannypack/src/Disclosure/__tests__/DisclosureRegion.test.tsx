import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Button } from '../../Button';
import { Blockquote } from '../../Blockquote';
import { Disclosure } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    function Component() {
      const hidden = Disclosure.useState({ baseId: 'test' });
      return (
        <div>
          <Disclosure {...hidden}>Toggle</Disclosure>
          <Disclosure.Region {...hidden} id="test">
            Hello world
          </Disclosure.Region>
        </div>
      );
    }
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly when the hidden is visible', () => {
    function Component() {
      const hidden = Disclosure.useState({ baseId: 'test', visible: true });
      return (
        <div>
          <Disclosure {...hidden}>Toggle</Disclosure>
          <Disclosure.Region {...hidden}>Hello world</Disclosure.Region>
        </div>
      );
    }
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('use', () => {
    it('should render correctly', () => {
      function Component() {
        const hidden = Disclosure.useState({ baseId: 'test' });
        return (
          <div>
            <Disclosure use={Button} {...hidden}>
              Toggle
            </Disclosure>
            <Disclosure.Region use={Blockquote} {...hidden}>
              Hello world
            </Disclosure.Region>
          </div>
        );
      }
      const { container } = render(<Component />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Disclosure props', () => {
      const { result } = renderHook(() => {
        const hidden = Disclosure.useState({ baseId: 'test' });
        return Disclosure.Region.useProps(hidden);
      });
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      function Component() {
        const hidden = Disclosure.useState({ baseId: 'test' });
        return (
          <Disclosure.Region {...hidden}>
            {DisclosureProps => <div {...DisclosureProps}>Hello world</div>}
          </Disclosure.Region>
        );
      }
      const { container } = render(<Component />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('Disclosure.root should render correctly', () => {
    const { container } = render(
      <Disclosure.Region
        baseId="test"
        overrides={{ Disclosure: { Region: { css: { root: { backgroundColor: 'red' } } } } }}
      >
        hello world
      </Disclosure.Region>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Disclosure.root should render correctly', () => {
    const { container } = render(<Disclosure.Region baseId="test">hello world</Disclosure.Region>, {
      theme: { Disclosure: { Region: { css: { root: { backgroundColor: 'red' } } } } }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Disclosure.Region baseId="test">hello world</Disclosure.Region>, {
      theme: { Disclosure: { Region: { defaultProps: { className: 'test' } } } }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
