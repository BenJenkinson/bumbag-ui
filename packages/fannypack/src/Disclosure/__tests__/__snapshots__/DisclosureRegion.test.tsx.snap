// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition hook should return with Disclosure props 1`] = `
Object {
  "className": "css-13h683e css-1pwrqjj css-3ekj6a fp-Box fp-DisclosureContent",
  "hidden": true,
  "id": "test",
  "onAnimationEnd": [Function],
  "onTransitionEnd": [Function],
  "style": Object {
    "display": "none",
  },
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-DisclosureContent"
  hidden=""
  id="test"
  style="display: none;"
>
  Hello world
</div>
`;

exports[`composition use should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 6px;
  color: #212121;
  fill: #212121;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  min-height: 2.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-hyphens: auto;
  -moz-hyphens: auto;
  -ms-hyphens: auto;
  hyphens: auto;
  -webkit-transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  transition: box-shadow 0.1s ease-in-out 0s,border 0.1s,background-color 0.1s;
  border: 1px solid #d9d9d9;
}

.emotion-0[disabled],
.emotion-0[aria-disabled="true"] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.emotion-0:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #b3b3b3 0px 0px 0px 1px, #d9d9d9 0px 0px 0px 3px;
  border-color: transparent;
}

.emotion-0:hover {
  background-color: #efefef;
}

.emotion-0:hover:active {
  background-color: #d6d6d6;
}

.emotion-4 {
  border-left: 4px solid #ccc;
  padding: 1rem;
}

.emotion-4:not(:last-child) {
  margin-bottom: 1rem;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="false"
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Button fp-Disclosure"
    type="button"
  >
    Toggle
  </button>
  <blockquote
    class="emotion-4 emotion-1 emotion-2 emotion-3 fp-Box fp-Blockquote fp-DisclosureContent"
    hidden=""
    id="test"
    style="display: none;"
  >
    Hello world
  </blockquote>
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-DisclosureContent"
  hidden=""
  id="test"
  style="display: none;"
>
  hello world
</div>
`;

exports[`overrides Disclosure.root should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  background-color: red;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-DisclosureContent"
  hidden=""
  id="test"
  style="display: none;"
>
  hello world
</div>
`;

exports[`props renders correctly when the hidden is visible 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="true"
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-Disclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-DisclosureContent"
    id="test"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

<div>
  <button
    aria-controls="test"
    aria-expanded="false"
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-Disclosure"
    type="button"
  >
    Toggle
  </button>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Box fp-DisclosureContent"
    hidden=""
    id="test"
    style="display: none;"
  >
    Hello world
  </div>
</div>
`;

exports[`theming Disclosure.root should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  background-color: red;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-DisclosureContent"
  hidden=""
  id="test"
  style="display: none;"
>
  hello world
</div>
`;
